R/sendmailR.r

```diff
@@ -169,12 +211,22 @@
 ##' body <- list("It works!", mime_part(iris))
 ##' sendmail(from, to, subject, body,
 ##'          control=list(smtpServer="ASPMX.L.GOOGLE.COM"))
+##' 
+##' sendmail(from="from@example.org",
+##'   to=c("to1@example.org", "to2@example.org"),
+##'   subject="SMTP auth test",
+##'   msg=mime_part("This message was send using sendmailR and curl."),
+##'   use_curl = TRUE,
+##'   curlopts = list(username = "foo", password = "bar"),
+##'   control=list(smtpServer="smtp://smtp.gmail.com:587", verbose = TRUE)
+##' )
 ##' }
-##'
 ##' @export
 sendmail <- function(from, to, subject, msg, cc, bcc, ...,
+                     use_curl = FALSE, use_ssl = "force",
```

Comment:
Since `use_ssl` is only used when `use_curl=TRUE`, should we move it to `curlopts()`?
More generally, should we anticipate any other engines and add a generic `engine` argument (with options `internal` (default)  and `curl`) plus `engineopts` for the engine specific options?

Reply:
I don't think we should anticipate other engines, but the `engine` argument is a much nicer interface. Since this approach is common practice in R, it is easier to understand for the user.

I renamed `curlopts` to engineopts. My first choice was `curlopts`, since this argument exposes a lot of curl functionality to the user. But I think users will realize this even with the new name :)